$MirOS: ports/www/firesomething/patches/patch-nsprpub_pr_src_misc_prinit_c,v 1.4 2008/12/05 16:26:11 bsiegert Exp $
$OpenBSD: patch-nsprpub_pr_src_misc_prinit_c,v 1.1.1.1 2004/02/15 19:37:25 wilfried Exp $
--- mozilla/nsprpub/pr/src/misc/prinit.c.orig	2008-04-26 02:39:35.000000000 +0200
+++ mozilla/nsprpub/pr/src/misc/prinit.c	2008-08-24 14:52:51.000000000 +0200
@@ -429,6 +429,12 @@ PR_IMPLEMENT(PRStatus) PR_Cleanup()
 	    _PR_LogCleanup();
 
         /*
+         * accesses the current thread
+         */
+        _PR_CleanupNet();
+        _PR_CleanupIO();
+
+        /*
          * This part should look like the end of _PR_NativeRunThread
          * and _PR_UserRunThread.
          */
@@ -440,6 +446,7 @@ PR_IMPLEMENT(PRStatus) PR_Cleanup()
             PR_DELETE(me->stack);
             PR_DELETE(me);
         }
+        _PR_MD_SET_CURRENT_THREAD(NULL);
 
         /*
          * XXX: We are freeing the heap memory here so that Purify won't
